.github/workflows/main.yml

```diff
@@ -3,48 +3,59 @@ name: main
 on:
   push:
     branches:
-    - master
+      - master
+      - st4000-exploration
   pull_request:
     branches:
-    - master
+      - master
+      - st4000-exploration
+
+env:
+  CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
+  SUBLIME_TEXT_VERSION: "3"
+  SUBLIME_TEXT_ARCH: x64
+  PACKAGE: LSP
 
 jobs:
-  macOS:
-    runs-on: macOS-latest
-    env:
-      CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
-      SUBLIME_TEXT_VERSION: "3"
-      SUBLIME_TEXT_ARCH: x64
-      PACKAGE: LSP
+  Mac-test:
+    runs-on: macos-latest
+    if: github.ref == 'refs/heads/master' || github.base_ref == 'master'
+    steps:
+      - run: echo ::add-path::$HOME/.local/bin
+      - run: echo ::add-path::$HOME/Library/Python/3.7/bin
+      - uses: actions/checkout@v1
+      - run: python3 tests/server.py --version
+      - run: python3 tests/adjust_unittesting_config_for_ci.py && cat unittesting.json
+      - run: curl --silent https://raw.githubusercontent.com/SublimeText/UnitTesting/master/sbin/ci.sh --output ci.sh
+      - run: |
+          sh ci.sh bootstrap
+          sh ci.sh install_package_control
+          subl --stay &
+          sleep 8
+          sh ci.sh run_tests --coverage
+        name: Run UnitTesting tests
+      - run: |
+          pip3 install coverage==4.5.4 codecov==2.0.15
+          codecov
+
+  Mac-lint:
```

Comment:
Should we even run lint on both mac and Linux? I can't imagine the lint issues to be platform-specific.

Reply:
I've only left linting running on linux. Lemme know what you think.